WordPress/Classes/ViewRelated/Blog/BlogDetailHeaderView.m

```diff
@@ -47,39 +43,88 @@ - (void)setBlog:(Blog *)blog
-                                                                 metrics:metrics
-                                                                   views:views]];
-    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"|[_blavatarImageView(blavatarSize)]-labelHorizontalPadding-[_subtitleLabel]|"
-                                                                 options:0
-                                                                 metrics:metrics
-                                                                   views:views]];
-    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_titleLabel(labelHeight)][_subtitleLabel(labelHeight)]"
-                                                                 options:0
-                                                                 metrics:metrics
-                                                                   views:views]];
-    [super setNeedsUpdateConstraints];
+    UIStackView *stackView = [[UIStackView alloc] init];
+    stackView.translatesAutoresizingMaskIntoConstraints = NO;
+    stackView.axis = UILayoutConstraintAxisHorizontal;
+    stackView.distribution = UIStackViewDistributionFill;
+    stackView.alignment = UIStackViewAlignmentCenter;
+    stackView.spacing = BlogDetailHeaderViewLabelHorizontalPadding;
+    [self addSubview:stackView];
+
+    NSLayoutConstraint *leadingConstraint;
+    NSLayoutConstraint *trailingConstraint;
+
+    if ([WPDeviceIdentification isiPhone]) {
+        // On iPhone, the readable content guide seems to be accurately aligned with the cell's content margins.
+        UILayoutGuide *readableGuide = self.readableContentGuide;
+        leadingConstraint = [stackView.leadingAnchor constraintEqualToAnchor:readableGuide.leadingAnchor];
+        trailingConstraint = [stackView.trailingAnchor constraintEqualToAnchor:readableGuide.trailingAnchor];
+    } else {
+        // On iPad, the correct readable margins seem to already be inherited via the tableHeaderView
+        // so following this view's readable content guide is not necessary.
+        leadingConstraint = [stackView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor];
+        trailingConstraint = [stackView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor];
+    }
+
+    [NSLayoutConstraint activateConstraints:@[
+                                              leadingConstraint,
+                                              trailingConstraint,
+                                              [stackView.topAnchor constraintEqualToAnchor:self.topAnchor],
+                                              [stackView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor],
+                                              [stackView.centerXAnchor constraintEqualToAnchor:self.centerXAnchor],
+                                              [stackView.centerYAnchor constraintEqualToAnchor:self.centerYAnchor]
```

Comment:
I'm going to have to research this a bit as technically we don't know what the `x` and `y` position of the view is, which is where IB often complains without these constraints. However, in code I suppose it doesn't matter since the view actually does init with `CGRectZero` in which the position is known and where we want it.

So, I'm going to go ahead and remove these.